### PR TITLE
TYP : DataFrame.(merge, join)  core.reshape.merge.(merge, ...) (easy)

### PR DESCRIPTION
**Pull request by arw2019**
- [ ] closes #xxxx
- [ ] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by jreback** [comment]
looks good. @simonjayhawkins if you can double check here.

**Review by simonjayhawkins — APPROVED** [review]
Thanks @arw2019 lgtm

in `_join_compat`, `on` is passed onto `left_on` in `core.reshape.merge`.

presumably both `left_on` and `right_on` can also be typed as `Optional[IndexLabel]` (in DataFrame.merge, core.reshape.merge  and _MergeOperation)

File: pandas/core/frame.py [hunk 1]
```diff
@@ -8222,18 +8229,18 @@ def _join_compat(
...
-        right_index=False,
-        sort=False,
-        suffixes=("_x", "_y"),
-        copy=True,
-        indicator=False,
-        validate=None,
+        left_index: bool = False,
+        right_index: bool = False,
+        sort: bool = False,
+        suffixes: Tuple[str, str] = ("_x", "_y"),
```
[the remark this answers is not in the document]
Comment (arw2019) [reply pandas/core/frame.py:8240]: Is it worth creating a Suffix alias? (So there's a single place where the behavior can be changed?) I would also change this to List if doing that

[the remark this answers is not in the document]
Comment (simonjayhawkins) [reply pandas/core/frame.py:8240]: yes, probably worth creating an alias as used in several places.

I don't think we need to include Sequence though (situation maybe different if types were public/not experimental or we were using a stub that had support for multiple pandas versions).

**Comment by jreback** [comment]
thanks @arw2019

**Pull request merged**